app/cli/internal/telemetry/telemetry.go

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"context"
+	"runtime"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/internal/attestation/crafter"
+	"github.com/denisbrodbeck/machineid"
+	"github.com/rs/zerolog"
+)
+
+const commandTrackerEventName = "command_executed"
+
+// Tags represents a collection of event tags.
+type Tags map[string]string
+
+// Client defines the interface for tracking events.
+type Client interface {
+	TrackEvent(ctx context.Context, eventName string, id string, tags Tags) error
+}
+
+// CommandTracker is an implementation in charge of tracking Commands events sent from the CLI.
+type CommandTracker struct {
+	client Client
+}
+
+// NewCommandTracker creates a new CommandTracker instance.
+func NewCommandTracker(client Client) *CommandTracker {
+	return &CommandTracker{
+		client: client,
+	}
+}
+
+// Track sends a command event to the telemetry.Client.
+func (t *CommandTracker) Track(ctx context.Context, cmd string, tags Tags) error {
+	if t.client == nil {
+		return nil
+	}
+
+	computedTags := loadDefaultTags()
+	for k, v := range tags {
```

Comment:
add a comment here saying that we'll append the custom tags. We probably should check that we are not overriding and skip in those cases.

Reply:
Sure thing